crates/xline/src/conflict/uncommitted_pool.rs

```diff
@@ -0,0 +1,254 @@
+use std::collections::{hash_map, HashMap};
+
+use curp::server::conflict::CommandEntry;
+use curp_external_api::conflict::{ConflictPoolOp, UncommittedPoolOp};
+use itertools::Itertools;
+use utils::interval_map::IntervalMap;
+use xlineapi::{
+    command::{get_lease_ids, Command},
+    interval::BytesAffine,
+};
+
+use super::{filter_kv, intervals, is_xor_cmd};
+
+/// Uncommitted pool for KV commands.
+#[derive(Debug, Default)]
+pub(crate) struct KvUncomPool {
+    /// Interval map for keys overlap detection
+    map: IntervalMap<BytesAffine, Commands>,
+}
+
+impl ConflictPoolOp for KvUncomPool {
+    type Entry = CommandEntry<Command>;
+
+    fn remove(&mut self, entry: Self::Entry) {
+        let Some(entry) = filter_kv(entry) else {
+            return;
+        };
+        let intervals = intervals(&entry);
+        for interval in intervals {
+            if self
+                .map
+                .get_mut(&interval)
+                .map_or(false, |m| m.remove_cmd(&entry))
```

Comment:
Why do you remove command entry in the command? I think removing the interval from the IntervalMap is enough.

Reply:
There may be multiple commands with the same interval. We need to maintain a vec for all these commands.